app/controllers/users_controller.rb

```diff
@@ -0,0 +1,46 @@
+# continue
+class UsersController < ApplicationController
+ before_action :authenticate_user!, only: [:index, :update]
+
+  def index
+   if current_user.role.name == "admin"
+     @user = User.all
+     else
+      redirect_to root_path, alert: "Unauthorized access"
+    end
+  end
+
+  def new
+    @user = User.new
+
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
```

Comment:
Why we have not used `before_action` to DRY this finder method?

Reply:
Updated